Injector subclasses are incompatible with doc test.

```python
/usr/local/lib/python3.7/doctest.py:932: in find
    self._find(tests, obj, name, module, source_lines, globs, {})
.venv/lib/python3.7/site-packages/_pytest/doctest.py:406: in _find
    self, tests, obj, name, module, source_lines, globs, seen
/usr/local/lib/python3.7/doctest.py:991: in _find
    if ((inspect.isroutine(inspect.unwrap(val))
.venv/lib/python3.7/site-packages/_pytest/doctest.py:377: in _mock_aware_unwrap
    return real_unwrap(obj, stop=_is_mocked)
/usr/local/lib/python3.7/inspect.py:511: in unwrap
    while _is_wrapper(func):
/usr/local/lib/python3.7/inspect.py:505: in _is_wrapper
    return hasattr(f, '__wrapped__') and not stop(f)
.venv/src/dependencies/src/dependencies/_injector.py:69: in __getattr__
    raise DependencyError(message)
E   dependencies.exceptions.DependencyError: 'Injector' can not resolve attribute '__wrapped__'
```

--- a/.coveragerc
+++ b/.coveragerc
@@ -4,6 +4,7 @@ source =
   _dependencies
   dependencies
   django_project
+  doctested_module
   examples
   flask_project
   helpers

--- a/src/_dependencies/injector.py
+++ b/src/_dependencies/injector.py
@@ -41,7 +41,10 @@ class InjectorType(InjectorTypeType):
     def __getattr__(cls, attrname):
         __tracebackhide__ = True
 
-        cache, cached = {"__self__": cls}, {"__self__"}
+        cache, cached = (
+            {"__self__": cls, "__wrapped__": None},
+            {"__self__", "__wrapped__"},
+        )
         current_attr, attrs_stack = attrname, [attrname]
         have_default = False
 

--- a/tests/helpers/doctested_module.py
+++ b/tests/helpers/doctested_module.py
@@ -1,0 +1,21 @@
+# -*- coding: utf-8 -*-
+from dependencies import Injector
+
+
+class Foo(object):
+    def __init__(self, v):
+        self.v = v
+
+
+class Container(Injector):
+    v = 1
+    foo = Foo
+
+
+def container():
+    """
+    >>> container()
+    1
+
+    """
+    return Container.foo.v

--- a/tests/helpers/mddoctest.py
+++ b/tests/helpers/mddoctest.py
@@ -1,4 +1,5 @@
 # -*- coding: utf-8 -*-
+import doctest
 import sys
 from doctest import testfile
 from glob import glob
@@ -18,6 +19,12 @@ def _setup():
     responses.start()
 
 
+def _advanced_test():
+    import doctested_module
+
+    doctest.testmod(doctested_module)
+
+
 def _main():
     markdown_files = glob("**/*.md", recursive=True)
     exit_code = 0
@@ -29,4 +36,5 @@ def _main():
 
 if __name__ == "__main__":
     _setup()
+    _advanced_test()
     _main()

--- a/tests/helpers/pyproject.toml
+++ b/tests/helpers/pyproject.toml
@@ -5,6 +5,7 @@ description = ""
 authors = []
 packages = [
     { include = "django_project" },
+    { include = "doctested_module.py" },
     { include = "examples" },
     { include = "flask_project" },
     { include = "helpers.py" },

